### PR TITLE
NEXUS-23433 move orient dependent class into orient specific package

### PR DESCRIPTION
**Pull request by mickoallen**
JIRA: https://issues.sonatype.org/browse/NEXUS-23433

### CODE REVIEW

**Comment by mickoallen** [comment]
Hi @igorudovika thanks for looking at this. As described in the ticket - https://issues.sonatype.org/browse/NEXUS-23433 - this change is in preparation for doing newdb, but it is expected that disabling orient will break most things, including this plugin.

File: nexus-repository-conan/src/main/java/org/sonatype/repository/conan/internal/orient/hosted/ConanHostedHelper.java [hunk 1]
```diff
@@ -1,4 +1,16 @@
...
+ * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
+ *
+ * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
+ * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
+ *
+ * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
+ * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
+ * Eclipse Foundation. All other trademarks are the property of their respective owners.
+ */
+package org.sonatype.repository.conan.internal.orient.hosted;
```
[the remark this answers is not in the document]
Comment (mickoallen) [reply nexus-repository-conan/src/main/java/org/sonatype/repository/conan/internal/orient/hosted/ConanHostedHelper.java:13]: good catch, this is a mistake

File: nexus-repository-conan/src/main/java/org/sonatype/repository/conan/internal/orient/package-info.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+ * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
+ *
+ * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
+ * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
+ * Eclipse Foundation. All other trademarks are the property of their respective owners.
+ */
+@FeatureFlag(name = "nexus.orient.store.content")
+package org.sonatype.repository.conan.internal.orient;
+
+import org.sonatype.nexus.common.app.FeatureFlag;
```
[the remark this answers is not in the document]
Comment (mickoallen) [reply nexus-repository-conan/src/main/java/org/sonatype/repository/conan/internal/orient/package-info.java:16]: added

File: nexus-repository-conan/src/main/java/org/sonatype/repository/conan/internal/orient/proxy/v1/ConanProxyControllerV1.java [hunk 3]
```diff
@@ -10,14 +10,15 @@
...
+package org.sonatype.repository.conan.internal.orient.proxy.v1;
 
 import javax.inject.Inject;
 import javax.inject.Named;
 import javax.inject.Singleton;
 import org.sonatype.nexus.repository.view.Router;
-import org.sonatype.repository.conan.internal.common.v1.ConanControllerV1;
+import org.sonatype.repository.conan.internal.orient.common.v1.ConanControllerV1;
 import org.sonatype.repository.conan.internal.common.v1.ConanRoutes;
+import org.sonatype.repository.conan.internal.proxy.v1.ConanProxyHandlers;
```
Comment (mpiggott) [inline nexus-repository-conan/src/main/java/org/sonatype/repository/conan/internal/orient/proxy/v1/ConanProxyControllerV1.java:21]: Should `CoanProxyHandlers` also be moved? The hosted equivalent one was.

Comment (mickoallen) [reply nexus-repository-conan/src/main/java/org/sonatype/repository/conan/internal/orient/proxy/v1/ConanProxyControllerV1.java:21]: There is no dependency on orient there, but i'm assuming when the newdb implementation is done things will shift around a bit anyway

Comment (mpiggott) [reply nexus-repository-conan/src/main/java/org/sonatype/repository/conan/internal/orient/proxy/v1/ConanProxyControllerV1.java:21]: > There is no dependency on orient there, but i'm assuming when the newdb implementation is done things will shift around a bit anyway

Unless I'm missing something it has the same level of dependency as `HostedHandlers`

**Pull request merged**